languages/js/src/Host.ts

```diff
@@ -117,38 +138,56 @@ export class Host {
 
   /**
    * Get user type for `cls`.
+   *
+   * @param cls Class or class name.
    */
-  getType(cls: Class): UserType | undefined {
-    const typ = this.types.get(cls);
-    if (typ === undefined) return undefined;
-
-    if (typ.id === undefined) {
-      throw new Error('invariant: class must be in names and ids.');
-    }
-
-    return typ;
+  getType(cls: Class | string): UserType | undefined {
+    return this.types.get(cls);
   }
 
   /**
    * Return user types that are registered with Host.
    */
-  *distinctUserTypes(): IterableIterator<UserType> {
+  private *distinctUserTypes(): IterableIterator<UserType> {
```

Comment:
what's the * do?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*

Reply:
huh, TIL 👍